### ISSUE TITLE
[Screen Reader-Microbit editor]: 'Alt' Text is not defined for the image.

### ISSUE THREAD

**Issue by v-swasha** [Microsoft/pxt#2585]
**User Experience:**
User with Visual disabilities might not able to understand the image if 'Alt' Text is not defined. Like in the scenario alt text is not defined for the images.

**Test Environment**
OS: RS2 Version 1703(OS Build 15063.483)
Platform: Edge.
Screen reader: Narrator

**Repro Steps**
1-Navigate to url:- https://makecode.com/acc
2-Navigate to the Microbit section element and select Code control given on it.
3-Navigate to the menu controls listed near the Simulator control .
4-Navigate to images and verify the issue by developer tool.

**Actual Result**
'Alt' Text is not defined for the images.

**Expected Result**
'Alt' text should be defined for the image.

**Note**
Alt text is not defined for all SVG images defined in the application.

**MAS Reference**
https://microsoft.sharepoint.com/teams/msenable/Documents/LCA/MAS%20-%20July%201,%202017%20Release/MAS%201.1.1%20%E2%80%93%20Non-text%20Content.docx?d=wd2d2051fbdc84af78e1838aae867e216&csf=1

**Suggested fix:**
In HTML developers can ,combine adjacent image and text links for the same resource, Use alt attributes on img elements , Use the body of the object element. 
e.g. 
img src="newsletter.gif" alt="Free newsletter.  
   Get free recipes, news, and more. Learn more." / 
 
object classid="http://www.example.com/animatedlogo.py"
  img src="staticlogo.gif" alt="Company Name" /
/object
**Key References for Fix:** 
http://www.w3.org/TR/UNDERSTANDING-WCAG20/text-equiv-all.html 

**Refer Attachment**
![mas 1 1 1_makecode_alt text is not defined for the image](https://user-images.githubusercontent.com/30459830/28766783-b398b754-75ee-11e7-87a3-4b1c7d49a12b.png)
#

**Comment by v-amik** [comment]
review done

**Comment by veler** [comment]
This is part of Blocky, a third-part component.

**Comment by abchatra** [comment]
We have got exception for Blockly bugs from Julia Liuson (CVP). Closing these as Won't Fix. Following is the reasoning.

----------------------

MakeCode can be used in either Text editor or Block editor.

Blocks editor is inherently visual and is based on google blockly. Usage requires drag and drop functionality and we don’t have a equivalence in MakeCode for this w.r.t accessibility. 
We have evaluated couple of research projects which tries to find a equivalence to block based coding for accessibility needs: 
•	Project Torino (Requires students to buy physical blocks)
•	Accessible blockly (This is text based coding without syntax errors)
Both these are drastic departure from the blockly and provides a complete new experience. These are still in research and we intend not to use them in MakeCode at this point of time.

As workaround, the text editor in MakeCode is the way we see students with the need of accessibility using MakeCode.
